src/fix_downleveled_decorators.ts

```diff
@@ -0,0 +1,55 @@
+import * as ts from "typescript";
+
+/**
+ * This fixes the downleveled decorators so Closure doesn't have
+ * trouble with them. The problem is that when `experimentalDecorators` is
+ * enabled, we TSC ends up converting a class decorator like this:
+ *
+ * @classDecorator
+ * export class Person { ... }
+ *
+ * to this:
+ *
+ * let Person = class Person { ... };
+ *
+ * as well as some calls to __decorate(Person, ...)
+ *
+ * The problem is that this causes Closure Compiler to fail with this error:
+ * ERROR - [JSC_CANNOT_CONVERT_YET] Transpilation of 'Classes with possible name shadowing' is not yet implemented.
+ * 21| let Person = class Person {
+ *                  ^^^^^^^^^^^^^^
+ *
+ * This transformer fixes the problem by converting the class expression
+ * to a class declaration.
+ */
```

Comment:
Can you be more specific with the before/after ast?
Isn't it more like
```
let Person = __decorate(class Person{...});
```
or 
```
let Person = class Person{...}
Person = __decorate(Person)
```

How does the assignment work if you converted the class expression to a decorator? I think removing the name from the class expression is more likely to work.

Maybe we need to write a test to make sure class decorators actually work, both compiled and uncompiled.

Reply:
Yeah, I'm planning on writing some tests to prove out this functionality as well as the import handling.

So this ends up transforming: 

```ts
@classDecorator
export class Person {
    private name_: string;
    constructor(name: string) {
        this.name_ = name;
    }

    @accessorDecorator(true)
    get name() {
        return this.name_;
    }
  
    @methodDecorator(true)
    greet() {
        console.log(`Hello, my name is ${this.name}.`);
    }
  }
```

into:

```js
class Person {
    name_;
    /**
     * @public
     * @param {string} name
     */
    constructor(name) {
        this.name_ = name;
    }
    /**
     * @public
     * @return {string}
     */
    get name() {
        return this.name_;
    }
    /**
     * @public
     * @return {void}
     */
    greet() {
        console.log(`Hello, my name is ${this.name}.`);
    }
};
__decorate([
    accessorDecorator(true)
], Person.prototype, JSCompiler_renameProperty("name", Person.prototype), null);
__decorate([
    methodDecorator(true)
], Person.prototype, JSCompiler_renameProperty("greet", Person.prototype), null);
/** @suppress {visibility} */
Person = __decorate([
    classDecorator
], Person);
export { Person };
/* istanbul ignore if */
if (false) {
    /**
     * @type {string}
     * @private
     */
    Person.prototype.name_;
}
```

Yeah I wonder if stripping the name from the `class` is really the right call and it'd end up just being something like

```js
let Person = class { }
```
instead.

Closure itself doesn't have any problem compiling it as is though.... no guarantees at the moment on it actually working though post-Closure compiler.

Reply:
I'm not sure how it'd work out.  I'm assuming that it's okay since that's what `tsickle` was doing before when downleveling decorators.  That being said, I'll see about getting some tests in place to prove it out.